dbos-rules.ts

```diff
@@ -348,28 +678,36 @@ function analyzeRootNode(eslintNode: EslintNode, eslintContext: EslintContext) {
   };
 
   const tsMorphNode = makeTsMorphNode(eslintNode);
+  if (testingValidityOfTestsLocally) checkDiagnostics(tsMorphNode);
 
   try {
     if (Node.isStatemented(tsMorphNode)) {
       tsMorphNode.getFunctions().forEach(analyzeFunction);
       tsMorphNode.getClasses().forEach(analyzeClass);
     }
     else {
-      throw new Error(`Was expecting a statemented node! Got this kind instead: ${tsMorphNode.getKindName()}`);
+      const possibleVersioningError = `\
```

Comment:
As discussed, we actually care about the version of the typescript compiler used by typescript-eslint, which constructs the AST. We should figure out a range of compatible versions (those using TS 5.4.x, for now) and fail if the app uses any version of typescript-eslint out of that range.